Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py

```diff
@@ -1032,7 +1032,47 @@ def checkout(self, argument, prune=None):
                 cwd=self.root_path,
             ).returncode else self.commit()
 
+        match = self.dev_branches.match(argument)
         branch_remote = self.remote_for(argument)
+
+        # if branch is not a dev branch and has a remote
+        if not match and branch_remote:
+            path = '{}/{}'.format(branch_remote, argument)
+            local_head = self.commit(branch=argument)
+            remote_head = self.commit(branch=path)
+            branches_for_local = self.branches_for(local_head.hash)
+            branches_for_remote = self.branches_for(remote_head.hash)
+
+            # this logic ensures that branches tracking the correct remote are not reset
+            # if the branches for both commits are the same, then they must be tracking the same remote branch
+            if set(branches_for_local) != set(branches_for_remote):
+                local_head_1_hash = run(
+                    [self.executable(), 'rev-parse', '{}~1'.format(argument)],
+                    cwd=self.root_path,
+                    capture_output=True,
+                    encoding='utf-8',
+                ).stdout.strip()
+                branches_for_local_1 = self.branches_for(local_head_1_hash)
+
+                # check if head~1 has remote branch, in case of local commit
+                if set(branches_for_local_1) != set(branches_for_remote):
+                    log.info("Local branch is not tracking the remote branch.")
+                    if Terminal.choose(
+                        "Local changes on {} will not be saved. Would you like to override the local version of this branch with the version from '{}'?".format(argument, path),
```

Comment:
We should guard this prompt with a flag that is `false` by default, and have `git-webkit checkout` pass the `true`. This function is potentially used in circumstances where the user wouldn't expect to be prompted, perhaps because the "user" is actually a machine.